plugins/out_s3/s3.c

```diff
@@ -1114,11 +1164,28 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
             flb_free(payload_buf);
         }
-        m_upload->upload_errors += 1;
         /* re-add chunk to list */
         if (chunk) {
-            s3_store_file_unlock(chunk);
             chunk->failures += 1;
+            if (chunk->failures > ctx->ins->retry_limit) {
+                s3_retry_warn(ctx, (char *) chunk->fsf->meta_buf, m_upload->input_name,
+                              chunk->create_time, FLB_FALSE);
+                s3_store_file_delete(ctx, chunk);
+                /* destory multipart upload if part_number is 1, else set as complete*/
```

Comment:
spelling: `destory`

Reply:
also would be ideal if instead of just saying what the code does, the comment explains why. Otherwise the comment brings no value so let's remove it.